tests/zkevm/test_worst_compute.py

```diff
@@ -741,6 +741,101 @@ def test_worst_unop(
     )
 
 
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("key_mut", [True, False])
+@pytest.mark.parametrize("val_mut", [True, False])
+def test_worst_tload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    key_mut: bool,
+    val_mut: bool,
+):
+    """Test running a block with as many TLOAD calls as possible."""
+    env = Environment()
+
+    start_key = 41
+    code_key_mut = Bytecode()
+    code_val_mut = Bytecode()
+    if key_mut and val_mut:
+        code_prefix = Op.PUSH1(start_key) + Op.JUMPDEST
+        code_key_mut = Op.POP + Op.GAS
+        code_val_mut = Op.TSTORE(Op.DUP2, Op.GAS)
+        loop_iter = Op.POP(Op.TLOAD(Op.DUP1))
+    if key_mut and not val_mut:
+        code_prefix = Op.JUMPDEST
+        loop_iter = Op.POP(Op.TLOAD(Op.GAS))
+    if not key_mut and val_mut:
+        code_prefix = Op.JUMPDEST
+        code_val_mut = Op.TSTORE(Op.CALLVALUE, Op.GAS)  # CALLVALUE configured in the tx
+        loop_iter = Op.POP(Op.TLOAD(Op.CALLVALUE))
+    if not key_mut and not val_mut:
+        code_prefix = Op.JUMPDEST
+        loop_iter = Op.POP(Op.TLOAD(Op.CALLVALUE))
```

Comment:
For the case of `start_key = 0` we could `code_prefix = Op.PC + Op.JUMPDEST` and `loop_iter = Op.TLOAD()`. Since `TLOAD` will always push 0 we thus dont need the `CALLVALUE` + `POP` opcodes.

Reply:
It seems to me: `value=start_key if not key_mut and val_mut else 0,` that this value is thus 0 for this specific part of the test? It can thus be changed to the suggestion above.

Reply:
Not sure I understand your comment. There's no case where `start_key=0`.
This case is about a stable key. This is passed as a `value` in the tx, and here used with `CALLVALUE` (since is a fixed value).